tasks/api/src/main/java/org/infinispan/tasks/TaskContext.java

```diff
@@ -50,7 +51,12 @@ public TaskContext marshaller(Marshaller marshaller) {
     * other caches can be obtained from the cache manager
     */
    public TaskContext cache(Cache<?, ?> cache) {
-      this.cache = Optional.of(cache);
+      Optional<Cache<?, ?>> optionalCache = Optional.of(cache);
```

Comment:
Optional is so so disgusting :(

Reply:
Removing pre-existing Optional was out of scope of this PR

Reply:
I know... I retract my disgustment. Yuk. :)